pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -65,45 +88,101 @@ func CreateVaultResources(vaultParams ResourceCreationOpts) (*string, *string, *
 
-	log.Logger().Debugf("Generating the vault cloudformation template")
+	log.Logger().Debugf("Generating the vault CloudFormation template")
 
 	// and also the YAML AWS CloudFormation template
 	yaml, err := template.JSON()
 	if err != nil {
-		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault cloudformation template failed")
+		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault CloudFormation template failed")
 	}
 
-	log.Logger().Debugf("Generated the vault cloudformation template successfully")
+	log.Logger().Debugf("Generated the vault CloudFormation template successfully")
 
 	yamlProcessed := string(yaml)
 	yamlProcessed = setProperIntrinsics(yamlProcessed)
 
 	// Create dynamic stack name
 	stackName := stackNamePrefix + suffixString
 
-	runCloudformationTemplate(&yamlProcessed, &stackName)
+	err = runCloudFormationTemplate(&yamlProcessed, &stackName, nil)
+	if err != nil {
+		return nil, nil, nil, nil, nil, errors.Wrap(err, "there was a problem running the Vault CloudFormation stack")
+	}
 
 	kmsID, err := getKmsID(awsKmsKey, stackName)
 	if err != nil {
-		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault cloudformation template failed")
+		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault CloudFormation template failed")
 	}
 
-	accessKey, keySecret, err := createAccessKey(vaultParams.Region, iamUsername)
+	accessKey, keySecret, err := createAccessKey(iamUsername)
 	if err != nil {
-		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault cloudformation template failed")
+		return nil, nil, nil, nil, nil, errors.Wrapf(err, "Generating the vault CloudFormation template failed")
 	}
 
 	return accessKey, keySecret, kmsID, s3Name, &dynamoDbTableName, nil
 }
 
+// CreateVaultResourcesBoot creates required Vault resources in AWS for a boot cluster
+func CreateVaultResourcesBoot(vaultParams ResourceCreationOpts) (*string, *string, *string, *string, *string, error) {
+	log.Logger().Infof("Creating vault resources with following values, %s, %s, %s, %s",
+		util.ColorInfo(vaultParams.Region),
+		util.ColorInfo(vaultParams.Domain),
+		util.ColorInfo(vaultParams.Username),
+		util.ColorInfo(vaultParams.TableName))
+
+	templatePath := filepath.Join(vaultParams.AWSTemplatesDir, vaultCloudFormationTemplateName)
+	exists, err := util.FileExists(templatePath)
+	if err != nil || !exists {
+		return nil, nil, nil, nil, nil, errors.Wrap(err, "there was a problem loading the vault_cf_tmpl.yml file")
+	}
+	templateBytes, err := ioutil.ReadFile(templatePath)
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+	stackName := stackNamePrefix + vaultParams.UniqueSuffix
+	err = runCloudFormationTemplate(aws.String(string(templateBytes)), aws.String(stackName), []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(resourceSuffixParamName),
+			ParameterValue: aws.String(vaultParams.UniqueSuffix),
```

Comment:
If it gets to this point either:

a) There's no vault installed previously
b) No existing resources have been provided

The suffix is indeed creating a new set of independent resources and existing resources can be provided so we don't have to create new ones.

Reply:
Thanks for clarifications. 

What will happen in the following scenario:
Fist run of boot:
- resources are created for a cluster 
- vault is created with these resources

vault gets somehow deleted. The boot is run second time with the same cluster name. Does the stack creation is executed?

I believe that we save the resource names in the requirements file and they get automatically reused by the second run, and the CloudFormation stack is not applied again. 

It might be that one of these resources doesn't exist anymore in AWS (e.g. was deleted or something happed). I think, currently we don't validate this by re-running the stack. This is why I was asking if is idempotent. 

We can handle this in a follow up. Maybe we can open an issue for it.